docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py

```diff
@@ -0,0 +1,51 @@
+import re
+
+import markdown_it
+
+_ALERT_PATTERN = re.compile(
+    r"^\[\!(TIP|NOTE|IMPORTANT|WARNING|CAUTION)\]\s*", re.IGNORECASE
+)
+
+
+def gfm_alert_to_admonition(md: markdown_it.MarkdownIt):
+    def gfm_alert_to_adm(state: markdown_it.rules_core.StateCore):
```

Comment:
I think we can make this a little less verbose with some imports;

```py
from markdown_it import MarkdownIt
from markdown_it.rules_core import StateCore
```

Maybe we could also add type annotations for the `[Token]` list below? [I think](https://markdown-it-py.readthedocs.io/en/latest/api/markdown_it.token.html) that'd be `from markdown_it.token import Token`